pyuvdata/uvdata/uvdata.py

```diff
@@ -689,6 +694,73 @@ def metadata_only(self):
 
         return metadata_only
 
+    def _set_future_array_shapes(self):
+        """
+        Set future_array_shapes to True and adjust required parameters.
+
+        This method should not be called directly by users; instead it is called
+        by file-reading methods and `use_future_array_shapes` to indicate the
+        `future_array_shapes` is True and define expected parameter shapes.
+
+        """
+        self.future_array_shapes = True
+        self._freq_array.form = ("Nfreqs",)
+        self._channel_width.form = ("Nfreqs",)
+        if not self.metadata_only:
+            for param_name in self._data_params:
+                getattr(self, "_" + param_name).form = ("Nblts", "Nfreqs", "Npols")
+
+    def use_future_array_shapes(self):
+        """
+        Change the array shapes of this object to match the planned future shapes.
+
+        This method sets allows users to convert to the planned array shapes changes
+        before the changes go into effect. This method sets the `future_array_shapes`
+        parameter on this object to True.
+
+        """
+        self._set_future_array_shapes()
+        if not self.metadata_only:
+            for param_name in self._data_params:
+                setattr(self, param_name, (getattr(self, param_name))[:, 0, :, :])
+
+        self.freq_array = self.freq_array[0, :]
+
+        self.channel_width = np.zeros(self.Nfreqs, dtype=np.float) + self.channel_width
+
+    def use_current_array_shapes(self):
+        """
+        Change the array shapes of this object to match the current future shapes.
+
+        This method sets allows users to convert back to the current array shapes.
+        This method sets the `future_array_shapes` parameter on this object to False.
+        """
+        if self.Nspws == 1:
```

Comment:
I think this needs handling for `flex_spw` explicitly -- one can have a single SPW that is irregularly sampled/varying channel widths (i.e., due to flagging of RFI). In that case, I think one would handle this array in the identical way that one would deal with `freq_array`.

Reply:
Ok, I think this is a case where I was conflating `flex_spw` with `Nspws>1`, I didn't think about having one spw with variable channel widths within it.